confluent_kafka/src/Consumer.c

```diff
@@ -391,12 +391,18 @@ static PyObject *Consumer_poll (Handle *self, PyObject *args,
 
 static PyObject *Consumer_close (Handle *self, PyObject *ignore) {
 	CallState cs;
+	int raise = 0;
 
 	CallState_begin(self, &cs);
 
 	rd_kafka_consumer_close(self->rk);
 
-	if (!CallState_end(self, &cs))
+	raise = !CallState_end(self, &cs);
```

Comment:
Yes, almost, dealloc() is eventually triggered by the GC and I left it for that to clean up any lingering objects (such as callbacks), but they will be cleaned up on the next GC spin.

We could call cleanup from close() as an optimization to speed up GC, apart from that there shouldnt be any functional differences.
What do you think?